tests/test_cli_compile.py

```diff
@@ -572,38 +594,42 @@ def test_default_index_url(pip_with_index_conf):
     assert expected in output
 
 
-def test_stdin_without_output_file(runner):
+def test_stdin_without_output_file(invoke_cli):
     """
     The --output-file option is required for STDIN.
     """
-    out = runner.invoke(cli, ["-n", "-"])
+    out = invoke_cli(["-n", "-"])
 
     assert out.exit_code == 2
     assert "--output-file is required if input is from stdin" in out.stderr
 
 
-def test_not_specified_input_file(runner):
+def test_not_specified_input_file(invoke_cli):
     """
     It should raise an error if there are no input files or default input files
     such as "setup.py" or "requirements.in".
     """
-    out = runner.invoke(cli)
+    out = invoke_cli()
     assert "If you do not specify an input file" in out.stderr
     assert out.exit_code == 2
 
 
-def test_stdin(runner):
+def test_stdin(invoke_cli, from_line, depcache_dir):
     """
     Test compile requirements from STDIN.
     """
-    out = runner.invoke(
-        cli, ["-", "--output-file", "requirements.txt", "-n"], input="small-fake-a==0.1"
+    dep_cache = DependencyCache(depcache_dir)
+    dep_cache[from_line("small-fake-a==0.1")] = []
```

Comment:
Better to use `pip_conf`fixture. This allows you to install test wheels.

Reply:
Ah, I hadn't understood the purpose of that fixture. That's great.